jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +91,92 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @param tempDirPath path to temporary jib local directory to use.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the jar file or if the directory for the
+   *     temporary directory path provided doesn't exist.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath, @Nullable Path tempDirPath)
+      throws IOException {
+    Path localExplodedJarRoot = tempDirPath;
+    try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider()) {
```

Comment:
Oh! I realized there's a problem. When `TempDirectoryProvider` is closed, all the contents (i.e., exploded jar) will be cleaned up. That is, nothing left when exiting the method. However, tar creating happens far later in jib-core. I guess this method should just use `tempDirPath` (should not be `@Nullable`), unless the method opts to use a system temp directory. Since the method can accept `tempDirPath`, maybe we should just use it.

And how about `tempDirPath` --> `tempDirectory`?